### ISSUE TITLE
Integrate Library/Install tabs

### ISSUE THREAD

**Issue by KairuByte** [Phalcode/gamevault-app#169]
**Is your feature request related to a problem? Please describe.**
When you have a game available in the library, and installed, you have an entry for it in each tab. This is going to be a little confusing for end users, and honestly seems semi useless.

**Describe the solution you'd like**
The installed tab adds virtually nothing to the UI in my opinion. It could easily be relegated to a single "play" button on top of the icon in the library tab.

Executable selection can be configured in the games library page for now, and in a properties window when the application is mature enough to need/have one.

**Describe alternatives you've considered**
I'm unsure about the play button in the libraries main view, but could not think of a more apt way to handle this.

**Additional context**
This also solves the issue of there being no search or filter in the Installed tab, and would simultaneously solve #11

**Comment by Alfagun74** [comment]
It might seem that way on a 10 game server, but once you join a 500 game server, you'll have an absolutely delightful time exploring the extensive library of your installed games. Enjoy your day to the fullest! 😄

If you compare it with Steam:
- Our Library is the Store
- Our Installations are the Library
- and our Downloads are the Downloads


The install tab is not at all useless and comes with some features:
- It displays only the games you have installed, without requiring a connection to the server
- You can select an executable for each game.
- It allows you to conveniently open the installation folder for modding and other purposes.
- It allows you to delete the installation

We would greatly appreciate it if you could kindly tell us on how to incorporate these features into the library, which:
- Always connects to the server.
- By default displays all games available on that server.

**Comment by KairuByte** [comment]
Mmm this might get long, and I preemptively apologize if my tone is off.

>    * Our Library is the Store

The store is a "pre-library" point where you are requesting authorization to access a game. GV does not have such a "pre-library" point. 

>    * Our Installations are the Library

The library is the library. Any games you have access to are listed here, installed or not. They are differentiated in the UI by changing the listing and game info pages.

>    * and our Downloads are the Downloads

Steam downloads are actually "part of the library" (though this has slightly changed in more recent versions,) but in the case of Steam they are really informational and not meant to be interacted with beyond resume/pause and priority alterations. The cache is always cleared on it's own, there's no need to go in there and manually delete the files once the game is ready to play. (I actually think the download section needs some work as well but I haven't focused my attention towards it yet.)

> The install tab is not at all useless and comes with some features:

In the current state, yes. But not with some tweaks:

>    * It displays only the games you have installed, without requiring a connection to the server

The library can easily handle this as well. Just add a "installed/available/playable/not only on the back end/whatever" filter to the UI. If a user searches, and they have the filter set, show them the results of their installed games with a popup that says "hey! There's more available on some back ends, do you want to see *all* results?"

I'll address the connection below.

>    * You can select an executable for each game.

This can be handled elsewhere. I actually believe a better place is in a "properties" section, similar to how Steam handles this. An end user will likely set this once (ideally not even once but thats a discussion for a different issue :P) and never touch it again. It's not something that needs to be on a main tab.

>    * It allows you to conveniently open the installation folder for modding and other purposes.

See the above, this should really be buried in a right click menu.

>    * It allows you to delete the installation

Again this can be handled in a right click menu and/or on the games library page itself. This is going to be a feature used exactly once per game, there's no reason to have it displayed prominently.

> * Always connects to the server.

Make it not? While I admit with multiple servers things become a tad more annoying, there's no reason the total library info of each back end can't be cached locally and held until the user logs out/removes themselves from that connection. This would likely also improve things for when the back end is only online *sometimes* and the user wants to queue up downloads for the next time the server is online.

> *    By default displays all games available on that server.

Add a filter like described above.

I honestly feel like the install tab is a solution looking for a problem. Look at how little is done with it... 

You can:
* Play the game
  * Will be done all the time
  * Can be done from the library just as easily
* Go to the games library page 
  * Will be done sometimes
  * Can be done from the library already
* Change the exe
  * Will be done once unless you mess something up
  * Can be prompted for once an install has been completed
  * Can likely be automated for that first time... eventually
  * Can be hidden away from the main UI (library page/context menu/properties page)
* Delete the game
  * Will be done once
  * Can be hidden away from the main UI (library page/context menu/properties page)

The only thing done routinely is play the game. That means that for 99% of interactions with the install tab, it exists to hit "Play Game"... And there's no reason it explicitly has to exist in that tab when it can easily be put on, or next to, the icon in the library.

**Comment by Alfagun74** [comment]
Merged into #201
